integration-tests/bats/keyless.bats

```diff
@@ -31,15 +31,15 @@ teardown() {
 
 @test "keyless: feature indexes and foreign keys" {
     run dolt sql -q "ALTER TABLE keyless ADD INDEX (c1);"
-    [ $status -ne 0 ]
+    [ $status -eq 0 ]
     [[ ! "$output" =~ "panic" ]] || false
 
-    run dolt sql -q "CREATE TABLE bad (a int, b int, INDEX (b));"
-    [ $status -ne 0 ]
+    run dolt sql -q "CREATE TABLE fine (a int, b int, INDEX (b));"
+    [ $status -eq 0 ]
     [[ ! "$output" =~ "panic" ]] || false
 
     run dolt sql -q "CREATE TABLE worse (a int, b int, FOREIGN KEY (b) REFERENCES keyless(c1));"
-    [ $status -ne 0 ]
+    [ $status -eq 0 ]
```

Comment:
Foreign keys don't work properly. I pulled this branch and tried out a deletion with a `CASCADE` reference option and it's throwing an error. We should probably get foreign keys solved before putting this in, just in case we have to change the way keyless index maps are stored.